src/vs/workbench/parts/debug/browser/debugEditorActions.ts

```diff
@@ -209,10 +211,97 @@ class ShowDebugHoverAction extends EditorAction {
 	}
 }
 
+class GoToBreakpointAction extends EditorAction {
+	constructor(private isNext, opts) {
+		super(opts);
+	}
+
+	public run(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void | TPromise<void, any> {
+		const debugService = accessor.get(IDebugService);
+		const editorService = accessor.get(IEditorService);
+		const currentUri = editor.getModel().uri;
+		const currentLine = editor.getPosition().lineNumber;
+		const allEnabledBreakpoints =
+			debugService.getModel().getBreakpoints()
+				.filter(bp => bp.enabled)
+				.sort((a, b) => {
+					if (this.isNext) {
```

Comment:
Can we do this in twice as less lines? seems like a lot of code is duplicated in the next 20 lines?

Reply:
The sorting is unnecessary in the end since debugModel is responsible for sorting its own breakpoints - https://github.com/Microsoft/vscode/blob/cdd8256102433bf52c2f5b05aea312775a34a669/src/vs/workbench/parts/debug/common/debugModel.ts#L1003-L1015